File: zad2.py
class Topping:
    def __init__(self, name, price):
        self.name = name
        self.price = price

class Pizza:
    def __init__(self, size, crust):
        self.size = size
        self.crust = crust
        self.toppings = []

        # cena bazowa zależnie od rozmiaru
        if size == "S":
            self.basePrice = 20
        elif size == "M":
            self.basePrice = 28
        elif size == "L":
            self.basePrice = 36
        else:
            self.basePrice = 28  # domyślnie M

    def AddTopping(self, t):
        self.toppings.append(t)

    def TotalPrice(self):
        sum = self.basePrice
        for topping in self.toppings:
            sum += topping.price
        return sum

    def PrintPizza(self):
        print(f"Pizza {self.size}, ciasto: {self.crust}")
        if len(self.toppings) == 0:
            print("Dodatki: brak")
        else:
            print("Dodatki:")
            for topping in self.toppings:
                print(f"- {topping.name} (+{topping.price:.2f} zł)")
        print(f"Cena pizzy: {self.TotalPrice():.2f} zł")

class Order:
    def __init__(self):
        self.pizzas = []

    def AddPizza(self, p):
        self.pizzas.append(p)

    def TotalOrderPrice(self):
        sum = 0
        for pizza in self.pizzas:
            sum += pizza.TotalPrice()
        return sum

    def PrintSummary(self):
        print("\n=== PODSUMOWANIE ZAMÓWIENIA ===")
        for i, pizza in enumerate(self.pizzas, 1):
            print(f"\n--- Pizza #{i} ---")
            pizza.PrintPizza()
        print(f"\nRAZEM DO ZAPŁATY: {self.TotalOrderPrice():.2f} zł")

def main():
    menu = [
        Topping("Ser extra", 4),
        Topping("Szynka", 5),
        Topping("Pieczarki", 4),
        Topping("Papryka", 4),
        Topping("Oliwki", 4),
        Topping("Salami", 6)
    ]

    order = Order()

    print("=== APLIKACJA DO ZAMAWIANIA PIZZY ===")
    ile = int(input("Ile pizz chcesz zamówić? "))

    for i in range(ile):
        print(f"\n--- Tworzenie pizzy #{i + 1} ---")

        size = input("Wybierz rozmiar (S/M/L): ").upper()
        crust = input("Wybierz ciasto (cienkie/grube): ").lower()

        pizza = Pizza(size, crust)

        print("\nDostępne dodatki:")
        for t, topping in enumerate(menu, 1):
            print(f"{t}. {topping.name} (+{topping.price:.2f} zł)")

        n = int(input("Ile dodatków chcesz dodać? "))

        for k in range(n):
            nr = int(input("Podaj numer dodatku: "))

            if 1 <= nr <= len(menu):
                pizza.AddTopping(menu[nr - 1])
            else:
                print("Błędny numer dodatku - pomijam.")

        order.AddPizza(pizza)

    order.PrintSummary()

if __name__ == "__main__":
    main()

File: zad2.cs
using System;
using System.Collections.Generic;

class Topping
{
    public string name;
    public double price;

    public Topping(string name, double price)
    {
        this.name = name;
        this.price = price;
    }
}

class Pizza
{
    public string size;
    public string crust;
    public double basePrice;
    public List<Topping> toppings;

    public Pizza(string size, string crust)
    {
        this.size = size;
        this.crust = crust;
        this.toppings = new List<Topping>();

        // cena bazowa zależnie od rozmiaru
        if (size == "S") basePrice = 20;
        else if (size == "M") basePrice = 28;
        else if (size == "L") basePrice = 36;
        else basePrice = 28; // domyślnie M
    }

    public void AddTopping(Topping t)
    {
        toppings.Add(t);
    }

    public double TotalPrice()
    {
        double sum = basePrice;
        for (int i = 0; i < toppings.Count; i++)
        {
            sum += toppings[i].price;
        }
        return sum;
    }

    public void PrintPizza()
    {
        Console.WriteLine($"Pizza {size}, ciasto: {crust}");
        if (toppings.Count == 0)
        {
            Console.WriteLine("Dodatki: brak");
        }
        else
        {
            Console.WriteLine("Dodatki:");
            for (int i = 0; i < toppings.Count; i++)
            {
                Console.WriteLine("- " + toppings[i].name + " (+" + toppings[i].price.ToString("0.00") + " zł)");
            }
        }
        Console.WriteLine("Cena pizzy: " + TotalPrice().ToString("0.00") + " zł");
    }
}

class Order
{
    public List<Pizza> pizzas;

    public Order()
    {
        pizzas = new List<Pizza>();
    }

    public void AddPizza(Pizza p)
    {
        pizzas.Add(p);
    }

    public double TotalOrderPrice()
    {
        double sum = 0;
        for (int i = 0; i < pizzas.Count; i++)
        {
            sum += pizzas[i].TotalPrice();
        }
        return sum;
    }

    public void PrintSummary()
    {
        Console.WriteLine("\n=== PODSUMOWANIE ZAMÓWIENIA ===");
        for (int i = 0; i < pizzas.Count; i++)
        {
            Console.WriteLine("\n--- Pizza #" + (i + 1) + " ---");
            pizzas[i].PrintPizza();
        }
        Console.WriteLine("\nRAZEM DO ZAPŁATY: " + TotalOrderPrice().ToString("0.00") + " zł");
    }
}

class Program
{
    static void Main()
    {
        List<Topping> menu = new List<Topping>();
        menu.Add(new Topping("Ser extra", 4));
        menu.Add(new Topping("Szynka", 5));
        menu.Add(new Topping("Pieczarki", 4));
        menu.Add(new Topping("Papryka", 4));
        menu.Add(new Topping("Oliwki", 4));
        menu.Add(new Topping("Salami", 6));

        Order order = new Order();

        Console.WriteLine("=== APLIKACJA DO ZAMAWIANIA PIZZY ===");
        Console.Write("Ile pizz chcesz zamówić? ");
        int ile = Convert.ToInt32(Console.ReadLine());

        for (int i = 0; i < ile; i++)
        {
            Console.WriteLine("\n--- Tworzenie pizzy #" + (i + 1) + " ---");

            Console.Write("Wybierz rozmiar (S/M/L): ");
            string size = Console.ReadLine().ToUpper();

            Console.Write("Wybierz ciasto (cienkie/grube): ");
            string crust = Console.ReadLine().ToLower();

            Pizza pizza = new Pizza(size, crust);

            Console.WriteLine("\nDostępne dodatki:");
            for (int t = 0; t < menu.Count; t++)
            {
                Console.WriteLine((t + 1) + ". " + menu[t].name + " (+" + menu[t].price.ToString("0.00") + " zł)");
            }

            Console.Write("Ile dodatków chcesz dodać? ");
            int n = Convert.ToInt32(Console.ReadLine());

            for (int k = 0; k < n; k++)
            {
                Console.Write("Podaj numer dodatku: ");
                int nr = Convert.ToInt32(Console.ReadLine());

                if (nr >= 1 && nr <= menu.Count)
                {
                    pizza.AddTopping(menu[nr - 1]);
                }
                else
                {
                    Console.WriteLine("Błędny numer dodatku - pomijam.");
                }
            }

            order.AddPizza(pizza);
        }

        order.PrintSummary();
    }
}
